F2Py.cmake

```diff
@@ -171,6 +171,13 @@ function (f2py_add_module target_name)
     PRIVATE ${F2PY_ADD_MODULE_COMPILE_DEFS})
   endif()  
   target_compile_options(${target_name} PRIVATE -cpp)
+  # Link dll statically in Windows
+  # It can be potentially a PROBLEM!!!
+  # But this is the only way found to build a working library on Windows.
+  # However, on Linux and MacOS, it throws an error at the linking stage.
+  if (WIN32)
+    target_link_libraries(${target_name} PUBLIC "-static")
```

Comment:
I think this is a hack-ish way to do static linking. Let's postpone this to the CI PR

Reply:
Ok. target_link_options can be used instead.